protocols/raft/src/main/java/io/atomix/protocols/raft/roles/LeaderRole.java

```diff
@@ -1125,13 +1159,28 @@ private void stepDown() {
     }
   }
 
+  /**
+   * Fails pending commands.
+   */
+  private void failPendingCommands() {
```

Comment:
Pending commands that haven’t yet been written to the Raft log has to be rejected when a leader steps down to force the client to retry on another leader.